### ISSUE TITLE
Add a way to pass the context CFLAGS to foreign build sandboxing

### ISSUE THREAD

**Issue by samoht** [ocaml/dune#3341]
Hi there,

I am trying to compile some C code, using the foreign build sandboxing. This is working nicely, but now I would like to be able to tweak the C flags (for instance to use the ones defined in the current workspace context).

Is there a way to do that properly?

I've tried various things, but I only succeeded to make `dune` explode:

```

**Comment by rgrinberg** [comment]
Just to clarify, you'd like to print the c_flags so that some script can process them?

**Comment by samoht** [comment]
> Just to clarify, you'd like to print the c_flags so that some script can process them?

Yes. I found a way by printing `%{cc}` and removing the first world, so it's not blocking (just a bit surprising that it failed like that).

**Comment by snowleopard** [comment]
@samoht Thanks for reporting! I can look into this issue but probably not this/next week.

**Comment by dibrinsofor** [comment]
Any luck with this?

**Comment by snowleopard** [comment]
I'm not aware of anyone working on this at the moment. (Alas, I never found time to look.)
